src/coordinator/functions/logical/or.go

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package logical
+
+import (
+	"github.com/m3db/m3db/src/coordinator/block"
+)
+
+func makeOrBuilder(
+	logicalNode *BaseNode,
+	lIter, rIter block.StepIter,
+) (block.Block, error) {
+	meta, err := combineMetadata(lIter.Meta(), rIter.Meta())
+	if err != nil {
+		return nil, err
+	}
+	missingIndices, combinedSeriesMeta := missing(
```

Comment:
if combineMetadata produces a different ordering than missing, then this will break. I wonder how we can enforce the same ordering for both ? Maybe consider doing all this in a single function